sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -452,6 +452,32 @@ def test_max_feature_regression():
     assert_true(deviance < 0.5, "GB failed with deviance %.4f" % deviance)
 
 
+def test_feature_importance_regression():
+    # Test to make sure GINI importance is calculated correctly
+    # Make sure that results are summed, then normalized to avoid
+    # over-weighting features from later stages
+    boston = datasets.load_boston()
+    X, y = shuffle(boston.data, boston.target, random_state=13)
+    X = X.astype(np.float32)
+    offset = int(X.shape[0] * 0.9)
+    X_train, y_train = X[:offset], y[:offset]
+
+    clf = GradientBoostingRegressor(n_estimators=500, min_samples_split=2,
+                                    max_depth=4, learning_rate=0.01, loss='ls',
+                                    random_state=1)
+    clf.fit(X_train, y_train)
+
+    feature_importance = clf.feature_importances_
+    feature_importance = 100.0 * (feature_importance /
+                                  feature_importance.max())
+    sorted_idx = np.argsort(feature_importance)
+
+    feature_order = ['CHAS', 'ZN', 'RAD', 'INDUS', 'B', 'AGE', 'TAX',
+                     'PTRATIO', 'CRIM', 'NOX', 'DIS', 'RM', 'LSTAT']
+
+    assert_equal(list(boston.feature_names[sorted_idx]), feature_order)
```

Comment:
use assert instead of assert equal.

use `tolist()` instead of the `list` constructor

`assert boston.feature_names[sorted_idx].tolist() == ordered_features`